x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.js

```diff
@@ -66,7 +66,13 @@ export class BulkUploader {
 
     // this is internal bulk upload, so filter out API-only collectors
     const filterThem = _collectorSet => _collectorSet.getFilteredCollectorSet(c => c.ignoreForInternalUploader !== true);
-    this._fetchAndUpload(filterThem(collectorSet)); // initial fetch
+
+    if (this._timer) {
+      clearInterval(this._timer);
+    } else {
+      this._fetchAndUpload(filterThem(collectorSet)); // initial fetch
+    }
+
     this._timer = setInterval(() => {
       this._fetchAndUpload(filterThem(collectorSet));
     }, this._interval);
```

Comment:
we'll likely want to wait for completion of _fetchAndUpload before restarting the interval.  this may make many requests, and could result in a fair amount of back pressure

Reply:
Yes, and Cloud already suffered from it once: https://github.com/elastic/kibana/issues/30058

Reply:
👍 opened https://github.com/elastic/kibana/issues/31498.  quick fix for patching and we can address that separately.